### PR TITLE
Kernel: Stop requiring working malloc for syscall.h includes

### LINKED ISSUE #13869: <syscall.h> shouldn't require malloc
When including `syscall.h` after making `malloc` unavailable an error is thrown.

### Why do you need this?

I would like to make progress adding serenity support to a a system call fuzzer named syzkaller (https://github.com/SerenityOS/serenity/issues/1025). I have a POC, [but it requires a few hacks that aren't upstreamable](https://github.com/HerrSpace/syzkaller/commit/799fbf7fc9ce3c034e77cbd1a9b2fd98f05c8a50).

[Syzkaller deliberately makes malloc unavailable to enhance test reproducibility](https://github.com/google/syzkaller/blob/master/executor/executor.cc#L59-L64). As this works fine for every other operating system supported by syzkaller, I assume serenity should not require malloc here either.

### Minimal Repro

dummy.cpp:
```cpp
#include <stdlib.h>
#define malloc do_not_use_malloc
#include <syscall.h>
```

### Compiler Output
```
$ ~/repos/upstream/serenity/Toolchain/Local/i686/bin/i686-pc-serenity-gcc -I ~/repos/upstream/serenity/Build/i686/Root/usr/include/ -std=c++20 -o ./dummy dummy.cpp

/Users/space/repos/upstream/serenity/Build/i686/Root/usr/include/AK/kmalloc.h: In function 'void* kmalloc_array(AK::Checked<unsigned int>, AK::Checked<unsigned int>)':
dummy.cpp:2:16: error: 'do_not_use_malloc' was not declared in this scope
    2 | #define malloc do_not_use_malloc
      |                ^~~~~~~~~~~~~~~~~
/Users/space/repos/upstream/serenity/Build/i686/Root/usr/include/AK/kmalloc.h: In function 'void* kmalloc_array(AK::Checked<unsigned int>, AK::Checked<unsigned int>, AK::Checked<unsigned int>)':
dummy.cpp:2:16: error: 'do_not_use_malloc' was not declared in this scope
    2 | #define malloc do_not_use_malloc
      |                ^~~~~~~~~~~~~~~~~
```

### PR DESCRIPTION
**Pull request by HerrSpace**
Fixes #13869

### CODE REVIEW

File: Kernel/API/SyscallString.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#pragma once
```
Comment (linusg) [inline Kernel/API/SyscallString.h:1]: This needs a license header, as the linter says :)

File: Kernel/API/SyscallString.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+#include <AK/StringView.h>
+#include <Kernel/API/Syscall.h>
+
+namespace Kernel {
+namespace Syscall {
```
Comment (linusg) [inline Kernel/API/SyscallString.h:7]: Please use C++17 nested namespaces

```suggestion
namespace Kernel::Syscall {
```

File: Kernel/API/SyscallString.h [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+// Seperate header so syscall.h doesn't depend on malloc
```
Comment (linusg) [inline Kernel/API/SyscallString.h:9]:
```suggestion
// Separate header so syscall.h doesn't depend on malloc.
```

File: Kernel/API/SyscallString.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        ENUMERATE_SYSCALLS(__ENUMERATE_SYSCALL)
+#undef __ENUMERATE_SYSCALL
+    default:
+        break;
+    }
+    return "Unknown"sv;
+}
+
+}
+}
```
Comment (linusg) [inline Kernel/API/SyscallString.h:27]:
```suggestion
}
```

**Review by linusg — CHANGES_REQUESTED** [review]

File: Kernel/API/SyscallString.h (hunk 1 above)
Comment (HerrSpace) [reply Kernel/API/SyscallString.h:8]: Ack! :)

**Comment by HerrSpace** [comment]
All fixed!

**Review by linusg — APPROVED** [review]

**Pull request merged**